simulation/traffic_simulator/include/traffic_simulator/entity/entity_base.hpp

```diff
@@ -120,6 +120,9 @@ class EntityBase
 
   virtual auto getBehaviorParameter() const -> traffic_simulator_msgs::msg::BehaviorParameter = 0;
 
+  virtual auto getVehicleParameters() const
+    -> const traffic_simulator_msgs::msg::VehicleParameters = 0;
```

Comment:
The base class EntityBase should only have interfaces that are common to all entities. And EntityBase should not know what type of base class it will be. This function is odd in that EntityBase knows that it will be the base class for the Vehicle type and that the Vehicle type has a property called VehicleParameters.
Therefore, please remove this function and change the code that currently uses this function to use the capability query idiom for EntityBase types.

Reply:
You're totally right. I wasn't sure why `getVehicleParameters()` was added as part of this PR at all - I only suspected that it might be the result of some early discussion with @pawellech1. I've removed these changes completely - they are not needed for `respawn`.